### PR TITLE
RYA-442 Implementing the Start and Stop Query interactors.

### PR DESCRIPTION
**Pull request by kchilton2**
## Description
Implemented DefaultStartQuery and DefaultStopQuery interactors. The InMemoryQueryRepository was also updated to refresh its cache if the underlying QueryChangeLog has been updated.

### Tests
Updated and added tests.

### Links
https://issues.apache.org/jira/browse/RYA-442

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/665/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/666/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/667/

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/entity/StreamsQuery.java [hunk 1]
```diff
@@ -60,17 +63,25 @@ public String getSparql() {
...
     public int hashCode() {
-        return Objects.hash(queryId, sparql);
+        return Objects.hash(queryId, sparql, isActive);
     }
 
     @Override
     public boolean equals(final Object o) {
         if(o instanceof StreamsQuery) {
             final StreamsQuery other = (StreamsQuery) o;
             return Objects.equals(queryId, other.queryId) &&
```
Comment (ejwhite922) [inline extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/entity/StreamsQuery.java]: Use EqualsBuilder

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultStartQuery.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.apache.rya.streams.api.exception.RyaStreamsException;
+import org.apache.rya.streams.api.interactor.StartQuery;
+import org.apache.rya.streams.api.queries.QueryRepository;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * TODO
```
Comment (ejwhite922) [inline extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultStartQuery.java]: fill in TODO

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryChange.java [hunk 3]
```diff
@@ -37,9 +37,11 @@
 @DefaultAnnotation(NonNull.class)
 public final class QueryChange implements Serializable {
     private static final long serialVersionUID = 1L;
+
     private final UUID queryId;
     private final ChangeType changeType;
     private final Optional<String> sparql;
+    private final Optional<Boolean> isActive;
```
Comment (ejwhite922) [inline extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryChange.java]: Why can't this just be a primitive boolean so a null value isn't possible?

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/entity/StreamsQuery.java (hunk 1 above)
Comment (kchilton2) [reply extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/entity/StreamsQuery.java]: Why? This requires less object construction, so I'm not going to do that.

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryChange.java (hunk 3 above)
Comment (kchilton2) [reply extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryChange.java]: Null objects are ambiguous and prone to NPEs, so I avoid using them.

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultStartQuery.java (hunk 2 above)
Comment (kchilton2) [reply extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultStartQuery.java]: Done.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/669/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/670/

**Pull request closed without merging**